source/server/admin/admin.h

```diff
@@ -244,6 +246,31 @@ class AdminImpl : public Admin,
     TimeSource& time_source_;
   };
 
+  /**
+   * Implementation of OverloadManager that is never overloaded. Using this instead of the real
+   * OverloadManager keeps the admin interface accessible even when the proxy is overloaded.
+   */
+  struct NullOverloadManager : public OverloadManager {
+    struct NullThreadOverloadState : public ThreadLocalOverloadState {
+      NullThreadOverloadState(Event::Dispatcher& dispatcher) : dispatcher_(dispatcher) {}
+      const OverloadActionState& getState(const std::string&) override { return inactive_; }
+
+      const OverloadActionState inactive_ = OverloadActionState::Inactive;
+      Event::Dispatcher& dispatcher_;
+    };
+
+    NullOverloadManager(Event::Dispatcher& dispatcher) : thread_local_overload_state_(dispatcher) {}
+    void start() override {}
+    ThreadLocalOverloadState& getThreadLocalOverloadState() override {
```

Comment:
const ThreadLocalOverloadState& since ThreadLocalOverloadState has no mutation methods as part of the interface.

Reply:
I don't want to make this const because I expect in the near future that we'll want non-const methods on here.

Reply:
Tell me more about said non-const methods.  I would expect OverloadManager to be the one doing all the updates to overload state resources.  I'm guessing that you're referring to methods to create scalable timers, which in fact would need to be non-const.

That said, I wonder if scalable timers should be more closely integrated with the Event::Dispatcher interface.

Reply:
Yep, my concern is around adding methods to create scalable timers. I think we can discuss them in a separate issue, though, and leave this as-is here.

Reply:
I think I'd prefer to make this const for now and then remove it down the line if we start needing non-const access, just so we don't accidentally leave this non-const.

Reply:
Finally gave it a try and it turns out that won't work since getState() is non-const. getState is non-const because it looks up a key in the internal state map and creates an entry for it if one doesn't exist.

Reply:
It is strange that getState returns a reference instead of a value.  State is a primitive type (an enum)

Reply:
The callers hold onto the reference to avoid extra map lookups